### PR TITLE
Add cache purge

### PR DESCRIPTION
**Pull request by bernardolins**

### CODE REVIEW

File: api/cache.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	}
+	manager, err := getManager(c)
+	if err != nil {
+		return err
+	}
+	count, err := manager.PurgeCache(c.Request().Context(), name, args)
+	if err != nil {
+		return err
+	}
+	return c.String(http.StatusOK, fmt.Sprintf("Path found and purged on %d servers", count))
```
Comment (nettoclaudio) [inline api/cache.go]: When no objects were purged, does it make sense saying "Path found"?

File: api/cache_test.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+			manager: &fake.RpaasManager{
+				FakePurgeCache: func(instanceName string, args rpaas.PurgeCacheArgs) (int, error) {
+					return 36, nil
+				},
+			},
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run("", func(t *testing.T) {
```
Comment (nettoclaudio) [inline api/cache_test.go]:
```suggestion
		t.Run(tt.description, func(t *testing.T) {
```

File: rpaas/k8s.go [hunk 3]
```diff
@@ -530,6 +533,27 @@ func (m *k8sRpaasManager) UnbindApp(ctx context.Context, instanceName string) er
...
+	}
+	if args.Path == "" {
+		return 0, ValidationError{Msg: "path is required"}
+	}
+	purgeCount := 0
+	for _, podStatus := range podMap {
+		if !podStatus.Running {
+			continue
+		}
+		if err = m.cacheManager.PurgeCache(podStatus.Address, args.Path, args.PreservePath); err != nil {
```
Comment (nettoclaudio) [inline rpaas/k8s.go]: You could log the current error at least.

File: rpaas/nginx/manager.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+func NewNginxManager() NginxManager {
+	return NginxManager{
+		managePort:    managePort(),
+		purgeLocation: purgeLocationMatch(),
+		client: http.Client{
+			Timeout: defaultPurgeTimeout,
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					InsecureSkipVerify: true,
```
Comment (nettoclaudio) [inline rpaas/nginx/manager.go]: You can drop this line (according to line 92, the request is ever made at plain HTTP).

Comment (bernardolins) [reply rpaas/nginx/manager.go]: Sure, I'll remove it!

File: api/cache.go (hunk 1 above)
Comment (bernardolins) [reply api/cache.go]: I think so, the message says "Path found and purged on 0 servers" when no objects are purged. However, I agree that we can return a better message. What do you think?

Comment (nettoclaudio) [reply api/cache.go]: I mean saying "Path found" makes no sense. What about only to say "Object purged on %s servers".

Comment (bernardolins) [reply api/cache.go]: Yeah, I think that makes a lot more sense. I'll change the message

**Pull request merged**